node/pkg/logscribeconsumer/app.go

```diff
@@ -0,0 +1,220 @@
+package logscribeconsumer
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"os"
+	"time"
+
+	errorsentinel "bisonai.com/orakl/node/pkg/error"
+	"bisonai.com/orakl/node/pkg/utils/request"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+)
+
+func New(options ...AppOption) (*App, error) {
+	consoleWriter := zerolog.ConsoleWriter{Out: os.Stdout}
+
+	c := &AppConfig{
+		StoreInterval:     DefaultLogStoreInterval,
+		Buffer:            DefaultBufferSize,
+		LogscribeEndpoint: DefaultLogscribeEndpoint,
+		Level:             DefaultMinimalLogStoreLevel.String(),
+	}
+	for _, option := range options {
+		option(c)
+	}
+
+	resp, err := request.RequestRaw(request.WithEndpoint(c.LogscribeEndpoint))
+	if err != nil || resp.StatusCode != http.StatusOK {
+		return nil, errorsentinel.ErrLogscribeConsumerEndpointUnresponsive
+	}
+
+	if c.Service == "" {
+		log.Error().Msg("Service not provided")
+		return nil, errorsentinel.ErrLogscribeConsumerServiceNotProvided
+	}
+	level, err := zerolog.ParseLevel(c.Level)
+	if err != nil {
+		log.Error().Msgf("Error parsing log level, falling back to default value: %s", DefaultMinimalLogStoreLevel.String())
+		level = DefaultMinimalLogStoreLevel
+	}
+
+	return &App{
+		StoreInterval:     c.StoreInterval,
+		buffer:            make(chan map[string]any, c.Buffer),
+		consoleWriter:     consoleWriter,
+		LogscribeEndpoint: c.LogscribeEndpoint,
+		Service:           c.Service,
+		Level:             int(level),
+	}, nil
+}
+
+func (a *App) Write(p []byte) (n int, err error) {
+	if len(p) == 0 {
+		return 0, nil
+	}
+
+	_, err = a.consoleWriter.Write(p)
+	if err != nil {
+		return 0, err
+	}
+
+	res, err := byte2Entry(p)
+	if err != nil {
+		return 0, err
+	}
+	if len(res) == 0 {
+		return 0, errorsentinel.ErrLogEmptyLogByte
+	}
+
+	if !a.isLogLevelValid(res) {
+		return 0, err
+	}
+
```

Comment:
this function doesn't return error, will be more natural this way
```suggestion
	if !a.isLogLevelValid(res) {
		return 0, nil
	}

```

Reply:
or you can make isLogLevelValid to return error and use it

Reply:
reverted back to returning `nil`